bin/git-jira-pr

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env ruby
+# encoding: UTF-8
+#
+# git-jira-pr --
+#
+#   Open a pull request for the current branch in your default browser
+#
+#   Assumes the branches are named
+#   <team>/<branch-title>-<story-id>
+#
+require 'rubygems'
+require 'jira'
+require 'optparse'
+require 'cgi'
+require 'term/ansicolor'
+require 'git-whistles/app'
+require 'git-whistles/jira'
+
+class App < Git::Whistles::App
+
+  BROWSERS               = %w(xdg-open open firefox iceweasel)
+  SAFE_QUERY_STRING_SIZE = 8000
+
+  def initialize
+    super
+
+    @client = Git::Whistles::Jira.new.get_client
+  end
+
+  def main(args)
+    super
+    parse_args!(args)
+
+    if args.count > 0
+      die "Too many arguments", :usage => true
+    end
+
+    if options.from == options.to
+      die "You cannot issue a pull request to the same branch (#{options.from})."
+    end
+
+    query = { }
+
+    # guess team name
+    if options.from =~ %r{^(\w+-\w+)/.*}
+      team, issue_id = $1.capitalize.split('-')
+    else
+      issue_id = team = nil
+    end
+
+    # guess title.
+    title = options.from.split('/').last.split(/[_-]/).delete_if { |word| word =~ /^\d+$/ }.join(' ').capitalize
+    query[:"pull_request[title]"] = team ? "#{team}: #{title}" : title
+
+    # add Pivotal infos
```

Comment:
Done, thx for catching that.

On Wed, Feb 10, 2016 at 3:51 AM, Pedro Cunha <notifications@github.com>
wrote:

> In bin/git-jira-pr
> <https://github.com/mezis/git-whistles/pull/50#discussion_r52435542>:
>
> > +    end
> > +
> > +    query = { }
> > +
> > +    # guess team name
> > +    if options.from =~ %r{^(\w+-\w+)/.*}
> > +      team, issue_id = $1.capitalize.split('-')
> > +    else
> > +      issue_id = team = nil
> > +    end
> > +
> > +    # guess title.
> > +    title = options.from.split('/').last.split(/[_-]/).delete_if { |word| word =~ /^\d+$/ }.join(' ').capitalize
> > +    query[:"pull_request[title]"] = team ? "#{team}: #{title}" : title
> > +
> > +    # add Pivotal infos
>
> maybe remove this comment ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mezis/git-whistles/pull/50/files#r52435542>.
>